realm/realm-library/src/androidTest/java/io/realm/TestHelper.java

```diff
@@ -46,6 +47,23 @@
 
 public class TestHelper {
 
+    public static class ExpectedCountCallback implements RealmCache.Callback {
```

Comment:
Like I mentioned at the bottom I don't like the fact we're using callback to communicate results (its a blocking call anyway so `invokeWithGlobalRefCount` should just return the value) this is starting to look like Javascript 😱